parsl/providers/ad_hoc/ad_hoc.py

```diff
@@ -0,0 +1,235 @@
+            # Bash would return until the streams are closed. So we redirect to a outs file
+            cmd = 'bash {0} &> {0}.out & \n echo "PID:$!" '.format(script_path)
+            retcode, stdout, stderr = channel.execute_wait(cmd, self.cmd_timeout)
+            for line in stdout.split('\n'):
+                if line.startswith("PID:"):
+                    remote_pid = line.split("PID:")[1].strip()
+                    job_id = remote_pid
+            if job_id is None:
+                logger.warning("Channel failed to start remote command/retrieve PID")
+        else:
+
+            try:
+                job_id, proc = channel.execute_no_wait('bash {0}'.format(script_path), self.cmd_timeout)
+            except Exception as e:
+                logger.debug("Channel execute failed for: {}, {}".format(channel, e))
+                raise
+
+        self.resources[job_id] = {'job_id': job_id,
+                                  'status': 'RUNNING',
+                                  'cmd': 'bash {0}'.format(script_path),
+                                  'channel': channel,
+                                  'remote_pid': remote_pid,
+                                  'proc': proc}
+
+        return job_id
+
+    def status(self, job_ids):
+        """ Get status of the list of jobs with job_ids
+
+        Parameters
+        ----------
+
+        job_ids : list of strings
+          List of job id strings
+
+        Returns
+        -------
+        list of status strings ['PENDING', 'COMPLETED', 'FAILED']
+        """
+        for job_id in job_ids:
+            channel = self.resources[job_id]['channel']
+            status_command = "ps --pid {} | grep {}".format(self.resources[job_id]['job_id'],
+                                                            self.resources[job_id]['cmd'].split()[0])
+            retcode, stdout, stderr = channel.execute_wait(status_command)
+            if retcode != 0 and self.resources[job_id]['status'] == 'RUNNING':
+                self.resources[job_id]['status'] == 'FAILED'
+
+        return [self.resources[job_id]['status'] for job_id in job_ids]
+
+    def cancel(self, job_ids):
+        """ Cancel a list of jobs with job_ids
+
+        Parameters
+        ----------
+
+        job_ids : list of strings
+          List of job id strings
+
+        Returns
+        -------
+        list of confirmation bools: [True, False...]
+        """
+        logger.debug(f"Cancelling jobs: {job_ids}")
+        rets = []
+        for job_id in job_ids:
+            channel = self.resources[job_id]['channel']
+            retcode, stdout, stderr = channel.execute_wait("kill -TERM -{}".format(
+                self.resources[job_id]['job_id']))
+            if retcode == 0:
+                rets.append(True)
+            else:
+                rets.append(False)
+            self.resources[job_id]['status'] == 'COMPLETED'
+        return rets
+
+    @property
+    def scaling_enabled(self):
+        return True
```

Comment:
does this being true or false change anything with init_blocks=max_blocks=min_blocks? My initial reaction was to say "this should be false" but maybe it doesn't change anything?

Reply:
if `scaling_enabled==True` then parsl will do scaling actions, otherwise it is assumed that the user is doing the scaling actions separately. So min/max/init blocks controls how it can scale, and this piece enables the scaling actions.

Reply:
I'll wait for #1310, update ad-hoc, and then re-review.

Reply:
this is gone in #1310

Reply:
... but it needs deleting from this provider as it is now dead code